### PR TITLE
permissions: Add TouchUserPermissions and use during user permissions sync

### PR DESCRIPTION
**Pull request by ryanslade**
Add TouchUserPermissions to PermsStore

Then, touch user permissions if we encounter error in syncUserPerms so that the failed user sync
is moved to the back of the sync queue.

Part of https://github.com/sourcegraph/customer/issues/993

## Test plan

Tests added

### CODE REVIEW

**Review by sashaostrikov — APPROVED** [review]
LGTM!

**Comment by ryanslade** [comment]
@sashaostrikov Thanks, I'm going to wait for another 👍 from @unknwon

File: enterprise/cmd/repo-updater/internal/authz/perms_syncer_test.go [hunk 1]
```diff
@@ -199,6 +199,109 @@ func TestPermsSyncer_syncUserPerms(t *testing.T) {
...
+	perms := edb.NewMockPermsStore()
+	perms.SetUserPermissionsFunc.SetDefaultHook(func(_ context.Context, p *authz.UserPermissions) error {
+		wantIDs := []int32{1, 2, 3, 4, 5}
+		assert.Equal(t, wantIDs, p.GenerateSortedIDsSlice())
+		return nil
+	})
+	perms.UserIsMemberOfOrgHasCodeHostConnectionFunc.SetDefaultReturn(true, nil)
+	perms.TouchUserPermissionsFunc.SetDefaultHook(func(ctx context.Context, i int32) error {
+		touchUserPermsCalled = true
+		return nil
```
Comment (unknwon) [inline enterprise/cmd/repo-updater/internal/authz/perms_syncer_test.go:262]: You can do it in a more modern way using [`mockrequire.Called`](https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+mockrequire.Called&patternType=literal).

File: enterprise/internal/database/perms_store_test.go [hunk 2]
```diff
@@ -891,7 +891,7 @@ func testPermsStore_TouchRepoPermissions(db database.DB) func(*testing.T) {
 			t.Fatal(err)
 		}
 
-		// Touch the permissions in an hour late
+		// Touch the permissions to an hour later
```
Comment (unknwon) [inline enterprise/internal/database/perms_store_test.go:894]: 😂

**Review by unknwon — APPROVED** [review]

File: enterprise/cmd/repo-updater/internal/authz/perms_syncer_test.go (hunk 1 above)
Comment (ryanslade) [reply enterprise/cmd/repo-updater/internal/authz/perms_syncer_test.go:262]: I knew I'd seen something like that in the code. I went looking for it but couldn't remember what it was called, thanks :)

**Pull request merged**